core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala

```diff
@@ -0,0 +1,106 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import enumeratum.{Enum, EnumEntry}
+import org.broadinstitute.dsde.workbench.google2.GKEModels.{KubernetesClusterId, KubernetesClusterName, NodepoolName}
+import org.broadinstitute.dsde.workbench.google2.KubernetesModels.KubernetesApiServerIp
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.KubernetesNamespaceName
+import org.broadinstitute.dsde.workbench.google2.{Location, MachineTypeName, NetworkName, SubnetworkName}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+/** Google Container Cluster statuses
+ *  see: https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.Status
+ */
+
+case class KubernetesCluster(id: KubernetesClusterLeoId,
+                             googleProject: GoogleProject,
+                             clusterName: KubernetesClusterName,
+                             location: Location,
+                             status: KubernetesClusterStatus,
+                             serviceAccountInfo: WorkbenchEmail,
+                             samResourceId: KubernetesClusterSamResource,
+                             auditInfo: AuditInfo,
+                             asyncFields: KubernetesClusterAsyncFields,
+                             namespaces: Set[KubernetesNamespaceName],
+                             labels: LabelMap,
+                             nodepools: Set[Nodepool]
+                            )
+
+object KubernetesCluster {
+  implicit class EnrichedKubernetesCluster(cluster: KubernetesCluster) {
+    def getGKEClusterId: KubernetesClusterId = KubernetesClusterId(cluster.googleProject, cluster.location, cluster.clusterName)
+  }
+}
+
+case class SaveKubernetesCluster(googleProject: GoogleProject,
```

Comment:
Would `SavedKubernetesCluster` read better?

Reply:
Its a verb, this class is used to call the `.save()` function on a cluster, since not all fields in a `KubernetesCluster` should be specified at creation time. Its analagous to the `SaveCluster` object we have. I'll move it to the `KubernetesClusterComponent` class so the intent is more clear